src/main/java/com/atlassian/bitbucket/jenkins/internal/model/BitbucketBuildStatus.java

```diff
@@ -0,0 +1,74 @@
+package com.atlassian.bitbucket.jenkins.internal.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class BitbucketBuildStatus {
+
+    private final String description;
+    private final String key;
+    private final String name;
+    private final BuildState state;
+    private final String url;
+
+    public BitbucketBuildStatus(String description, String key, String name, BuildState state, String url) {
```

Comment:
```suggestion
    private BitbucketBuildStatus(String description, String key, String name, BuildState state, String url) {
```

We normally have private constructors for classes with builders. As far as I can tell there is no use of this constructor apart from the builder, so let's make it private.